packages/ripple/index.js

```diff
@@ -0,0 +1,174 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import {MDCRippleFoundation, util} from '@material/ripple';
+
+const withRipple = (WrappedComponent) => {
+  class RippledComponent extends Component {
+
+    foundation_ = null;
+    adapter_ = null;
+    instance_ = null;
+
+    state = {
+      classList: new Set(),
+      style: {},
+    };
+
+    componentDidMount() {
+      if (!this.adapter_) {
+        throw new Error('You must call initRipple from the element\'s ' +
+          'ref prop to initialize the adapter for withRipple');
+      }
+      this.foundation_ = new MDCRippleFoundation(this.adapter_);
+      this.foundation_.init();
+    }
+
+    componentWillUnmount() {
+      if (this.foundation_) {
+        this.foundation_.destroy();
+      }
+    }
+
+    createAdapter = (instance) => {
+      const adapter = {
+        browserSupportsCssVars: () => util.supportsCssVariables(window),
+        isUnbounded: () => this.props.unbounded,
+        isSurfaceActive: () => this.state.isActivated,
+        isSurfaceDisabled: () => this.props.disabled,
+        addClass: (className) =>
+          this.setState({classList: this.state.classList.add(className)}),
+        removeClass: (className) => {
+          const {classList} = this.state;
+          classList.delete(className);
+          this.setState({classList});
+        },
+        registerDocumentInteractionHandler: (evtType, handler) =>
+          document.documentElement.addEventListener(evtType, handler, util.applyPassive()),
+        deregisterDocumentInteractionHandler: (evtType, handler) =>
+          document.documentElement.removeEventListener(evtType, handler, util.applyPassive()),
+        registerResizeHandler: (handler) => window.addEventListener('resize', handler),
+        deregisterResizeHandler: (handler) => window.removeEventListener('resize', handler),
+        updateCssVariable: this.updateCssVariable,
+        computeBoundingRect: () => instance.getBoundingClientRect(),
+        getWindowPageOffset: () => ({x: window.pageXOffset, y: window.pageYOffset}),
+      };
+
+      this.instance_ = instance;
```

Comment:
saving this.instance_ within a "create" method is weird. Its a side effect of the "createAdapter" method that is unintuitive

Reply:
Will remove